router/worker.go

```diff
@@ -988,47 +1003,59 @@ func (w *worker) postStatusOnResponseQ(respStatusCode int, payload json.RawMessa
 			status.JobState = jobsdb.Filtered.State
 		}
 		w.logger.Debugf("sending success status to response")
-		w.rt.responseQ <- workerJobStatus{userID: destinationJobMetadata.UserID, worker: w, job: destinationJobMetadata.JobT, status: status}
-	} else {
-		// Saving payload to DB only
-		// 1. if job failed and
-		// 2. if router job undergoes batching or dest transform.
-		if payload != nil && (w.rt.enableBatching || destinationJobMetadata.TransformAt == "router") {
-			if w.rt.reloadableConfig.savePayloadOnError.Load() {
-				status.ErrorResponse = routerutils.EnhanceJSON(status.ErrorResponse, "payload", string(payload))
-			}
+		w.rt.responseQ <- workerJobStatus{
+			userID:  destinationJobMetadata.UserID,
+			worker:  w,
+			job:     destinationJobMetadata.JobT,
+			status:  status,
+			payload: inputPayload,
 		}
-		// the job failed
-		w.logger.Debugf("Job failed to send, analyzing...")
-
-		if isJobTerminated(respStatusCode) {
-			status.JobState = jobsdb.Aborted.State
-			w.updateAbortedMetrics(destinationJobMetadata.DestinationID, status.WorkspaceId, status.ErrorCode, errorAt)
-			destinationJobMetadata.JobT.Parameters = misc.UpdateJSONWithNewKeyVal(destinationJobMetadata.JobT.Parameters, "stage", "router")
-			destinationJobMetadata.JobT.Parameters = misc.UpdateJSONWithNewKeyVal(destinationJobMetadata.JobT.Parameters, "reason", status.ErrorResponse) // NOTE: Old key used was "error_response"
-		} else {
-			status.JobState = jobsdb.Failed.State
-			if !w.rt.retryLimitReached(status) { // don't delay retry time if retry limit is reached, so that the job can be aborted immediately on the next loop
-				status.RetryTime = status.ExecTime.Add(nextAttemptAfter(status.AttemptNum, w.rt.reloadableConfig.minRetryBackoff.Load(), w.rt.reloadableConfig.maxRetryBackoff.Load()))
-			}
+		return
+	}
+	// Saving payload to DB only
+	// 1. if job failed and
+	// 2. if router job undergoes batching or dest transform.
+	if payload != nil && (w.rt.enableBatching || destinationJobMetadata.TransformAt == "router") {
+		if w.rt.reloadableConfig.savePayloadOnError.Load() {
+			status.ErrorResponse = routerutils.EnhanceJSON(status.ErrorResponse, "payload", string(payload))
 		}
+	}
+	// the job failed
+	w.logger.Debugf("Job failed to send, analyzing...")
```

Comment:
Please **always** use the `n` (non-sugared) logger even if you don't pass any parameters. We do need a linter for this, but in the meantime try to remember about these please so that we won't have to work too much when we have a linter in place.

```suggestion
	w.logger.Debugn("Job failed to send, analyzing...")
```